app/src/test/java/com/example/parkingsystem/mvp/presenter/ReservationPresenterTest.java

```diff
@@ -0,0 +1,135 @@
+    TimePickerDialog.OnTimeSetListener listenerTime;
+    @Mock
+    DatePickerDialog.OnDateSetListener listenerDate;
+    ReservationContract.ReservationViewContract view = mock(ReservationContract.ReservationViewContract.class);;
+    ReservationContract.ReservationModelContract model = new ReservationModel(ReservationInformationDB.getInstanceDB());
+    ReservationContract.ReservationPresenterContract presenter =new ReservationPresenter(model, view);
+
+    @Test
+    public void saveReservationInformationWithReservationOkTest() {
+        saveReservationDateTime();
+        presenter.saveReservationInformation(SECURITY_CODE, PLACE);
+        verify(view).finishActivity(model.getReservation(PLACE, SECURITY_CODE));
+        assertEqualsReservationFields();
+    }
+
+    private Calendar getCalendarDate(int year, int month, int dayOfMonth, int hour, int minute) {
+        Calendar calendar = Calendar.getInstance();
+        calendar.set(Calendar.DAY_OF_MONTH, dayOfMonth);
+        calendar.set(Calendar.MONTH, month);
+        calendar.set(Calendar.YEAR, year);
+        calendar.set(Calendar.HOUR_OF_DAY, hour);
+        calendar.set(Calendar.MINUTE, minute);
+        return calendar;
+    }
+
+    @Test
+    public void saveReservationInformationWithIncompleteReservationTest() {
+        presenter.saveReservationInformation(EMPTY_STRING, EMPTY_STRING);
+        verify(view).showError();
+        assertEquals(null, model.getReservation(EMPTY_STRING, EMPTY_STRING));
+    }
+
+    @Test
+    public void saveReservationInformationWithOverlapReservationTest() {
+        saveReservationDateTime();
+        model.saveReservation(SECURITY_CODE, PLACE);
+        saveReservationDateTime();
+        presenter.saveReservationInformation(SECURITY_CODE, PLACE);
+        verify(view).showOverlapMessage();
+        assertEqualsReservationFields();
+    }
+
+    @Test
+    public void saveReservationTimeTest() {
+        model.setStartDateAndTime(true);
+        model.saveDate(YEAR_START, MONTH_START, DAY_OF_MONTH_START);
+        presenter.saveReservationTime(HOUR_START, MINUTE_START);
+        verify(view).showOkDateAndTime(model.getSavedDateAndTime());
+        Reservation reservation = new Reservation(getCalendarDate(YEAR_START, MONTH_START, DAY_OF_MONTH_START, HOUR_START, MINUTE_START),
+                null,
+                null,
+                null);
+        assertEquals(reservation.getStartDateAndTimeFormated(), model.getSavedDateAndTime());
+    }
+
+    @Test
+    public void saveReservationDateTest() {
+        model.setStartDateAndTime(true);
+        presenter.saveReservationDate(YEAR_START, MONTH_START, DAY_OF_MONTH_START, listenerTime);
+        verify(view).showTimePicker(listenerTime);
+    }
+
+    @Test
+    public void createDateTest() {
+        presenter.createDate(listenerDate, true);
+        verify(view).showDatePicker(listenerDate);
+    }
+
+    private void saveReservationDateTime() {
+        model.setStartDateAndTime(true);
```

Comment:
as you are testing the presenter you should not call the model, just call the methods on the presenter that set these values in the model

Reply:
fixed